Setup Script

jjjoycing123 opened the pull request:
### Changes
* Adds `setup.py`
* Closes #9

jjjoycing123 committed:
This is the setup.py that pycharm generated.

jjjoycing123 committed:
Moved setup.py to scripts dir.

aokellermann commented on scripts/setup.py:
```diff
@@ -0,0 +1,13 @@
+setup(
+    name='securedrop',
+    version='1.0',
+    packages=['securedrop', 'securedrop.tests'],
```
Use `find_packages`.

jjjoycing123 replied:
Changed

aokellermann commented on scripts/setup.py:
```diff
@@ -0,0 +1,13 @@
+    name='securedrop',
+    version='1.0',
+    packages=['securedrop', 'securedrop.tests'],
+    url='https://github.com/aokellermann/securedrop.git',
```
Remove trailing `.git`. It should link to project homepage, not clone link.

jjjoycing123 replied:
Changed

aokellermann commented on scripts/setup.py:
```diff
@@ -0,0 +1,13 @@
+    packages=['securedrop', 'securedrop.tests'],
+    url='https://github.com/aokellermann/securedrop.git',
+    license='MIT License',
+    author='John Nay, Antony Kellermann, Robert Liani',
```
This should be the name of the person who "owns" the project, which is me since it is hosted on my github account.

jjjoycing123 replied:
Changed.

aokellermann commented on scripts/setup.py:
```diff
@@ -0,0 +1,13 @@
+    url='https://github.com/aokellermann/securedrop.git',
+    license='MIT License',
+    author='John Nay, Antony Kellermann, Robert Liani',
+    author_email='John_Nay@student.uml.edu',
```
This should be my email (use the one listed on my github account).

jjjoycing123 replied:
Changed.

aokellermann replied:
This should be aokellermann@gmail.com

jjjoycing123 replied:
Changed

aokellermann commented on scripts/setup.py:
```diff
@@ -0,0 +1,13 @@
+    author_email='John_Nay@student.uml.edu',
+    description='Secure file sharing command line application.',
+    python_requires='>=3.8',
+)
```
You need to put in an `install_requires` section.

jjjoycing123 replied:
Added the packages on my machine (for example 'pycryptodome>=3.9.9') in the intstall_requires section.

aokellermann replied:
I think this includes non-dependencies. You can generate dependency list with `pip freeze`.

jjjoycing123 replied:
Changed. Thanks for the pip freeze suggestion!

jjjoycing123 committed:
Changed setup.py per Antony's suggestions. Added install_requires with the packages on my machine for the current project.

jjjoycing123 committed:
List Contacts (#36)

* Implemented user registration

* Add Client class

* Move dir

* cd into directory during CI

* Add . to pythonpath

* Some print statements

* More unit tests and login/autologin implementation

* Implemented stage 3

* Some documentation

* Fix bug

* SSL keygen

* Added generation of client cert/key

* Fix

* Fix test script

* Add get_type method

* Added utils file

* Interim commit

* Register successful

* Fix testing

* Add trap

* Login working

* Login working

* Implemented socket reuse

* Increase reuse

* Add contact working

* Replace Process with Thread and fix some bugs

* Implemented socketserver in command

* Implemented socketserver for server

* Tornado

* SSL workinggit add securedrop/client_server_base.pygit add securedrop/client_server_base.py

* del

* Packets

* Small change

* Interim commit

* Working single threaded server

* Increased test stress

* 1-3 working

* Final touches

* Fixes

* Refctor

* Update __init__.py

* Remove utils

* Update binary

* Add server binary

* chmod

* Fix

* Sanity check

* rename some cryptic stuff

* Rename constants with uppercase

* Rename more shm

* Rename another resp

* Adding List_Contacts_Packets.py class and List_Contacts_Response_Packets.py class.

* Adjusted ListContactsPackets.

* Adjusted List_Contacts_Response_Packets.py

* Adjusted List_Contacts_Response_Packets.py

* Added code to list contacts. Client now requests a list of other users which the server delivers as a dictionary.

* Prints Contact dictionary in client.py

* Changed client server test.

* Update shebangs (#34)

Update shebangs for python 3 for us ubuntu users ;)


Long Live the Focal Fossa!

* Change __init__.py.

* Improved the following files per Antony's suggestion: client.py, List_Contacts_Packets.py, List_Contacts_Response_Packets.py, server.py

* Remove dropped connections

* Sanity check

* Adding List_Contacts_Packets.py class and List_Contacts_Response_Packets.py class.

* Adjusted ListContactsPackets.

* Adjusted List_Contacts_Response_Packets.py

* Adjusted List_Contacts_Response_Packets.py

* Added code to list contacts. Client now requests a list of other users which the server delivers as a dictionary.

* Prints Contact dictionary in client.py

* Changed client server test.

* Change __init__.py.

* Improved the following files per Antony's suggestion: client.py, List_Contacts_Packets.py, List_Contacts_Response_Packets.py, server.py

* Rebased with most recent Tornado

* Added test to client_server_test.py and cleaned up server.py and client.py.

* Updated client_server_test.py so tests execute in the right order. Updated client.py to print the dictionary.

* formatted the list contacts stdout output.

* Got the list contacts milestone working with the three filters

* added a test for the list command when user has not added other users.

* fix

* Cleaned up the code per the PR. See my comments

* Resolved merge conflicts.

* Formatted files.

* Updated files per PR suggestions.

* Changed code in get_contacts() function per Antony's suggestion.

Co-authored-by: Antony Kellermann <aokellermann@gmail.com>
Co-authored-by: BobbyBytes <robbie234@gmail.com>

BobbyBytes committed:
Exe args (#39)

## Get executable args from command line input
* Closes #39 
Note :  Added the args for verbose output, to be implemented in #5  
# Client
- -f --filename 
- -p --port
- -a --address
- -v --verbose

# Server
- -f --filename 
- -p --port
- -v --verbose

jjjoycing123 committed:
Used pip freeze to get project dependencies and updated the install_requires section per what the project requires.

jjjoycing123 committed:
Merge branch 'master' into John_Issue_9_Branch

jjjoycing123 committed:
Forgot comma after tornado.

aokellermann committed:
Merge branch 'master' into John_Issue_9_Branch

aokellermann committed:
Fix setup

aokellermann committed:
Chmod

aokellermann committed:
Fix

aokellermann committed:
more fix
